tests/transformers/test_transformers_model_export.py

```diff
@@ -2455,72 +2468,6 @@ def test_instructional_pipeline_with_prompt_in_output(model_path):
     assert "\n\n" in inference[0]
 
 
-@pytest.mark.skipif(not _IS_PIPELINE_DTYPE_SUPPORTED_VERSION, reason="Feature does not exist")
-@flaky()
-def test_load_as_pipeline_preserves_framework_and_dtype(model_path):
```

Comment:
@BenWilson2 The pipeline instance no longer holds `torch_dtype` instance attribute but just emits underlying model `dtype` after [my change](https://github.com/huggingface/transformers/pull/28940) in Transformers (as per the suggestion from their maintainer). Therefore, passing `torch_dtype` to pipeline constructor with a pre-instantiated model instance will not help saving the model with different `torch_dtype`. To continue supporting that use case, this PR adds `torch_dtype` argument to `save_model()` and `log_model()`.

The following shows the different in how to save model with different dtype, before and after this PR. No practical change in the behavior, but it differs in terms of where to pass the argument.

Before:
```
pipeline = transformers.pipeline(
    model=model,
    tokenizer=tokenizer,
    torch_dtype=torch.float16   # Different from model dtype
)
mlflow.transformers.save_model(
    transformers_model=pipeline,
    path="model",
)
```

After:
```
pipeline = transformers.pipeline(
    model=model,
    tokenizer=tokenizer,
)
mlflow.transformers.save_model(
    transformers_model=pipeline,
    path="model",
    torch_dtype=torch.float16   # Different from model dtype
)
```

Indeed I think the new interface is more intuitive (the previous one relies on a bit tricky fact that passing `torch_dtype` to pipeline constructor doesn't update the model's dtype), but please let me know if you think we should keep the original UX.

Reply:
This is more intuitive AND it's more correct for storing the attribute as part of the bottleneck of processing (the model). 
Let's verify this behavior change with Inference team.